lib/livebook/hubs/team_client.ex

```diff
@@ -357,7 +368,29 @@ defmodule Livebook.Hubs.TeamClient do
     |> dispatch_deployment_groups(user_connected)
   end
 
+  defp handle_event(:agent_connected, agent_connected, state) do
+    %{state | deployment_groups: [agent_connected.deployment_group]}
+    |> dispatch_secrets(agent_connected)
+    |> dispatch_file_systems(agent_connected)
+  end
+
+  defp dispatch_secrets(state, %{
+         deployment_group: %{secrets: agent_secrets},
+         secrets: org_secrets
+       }) do
+    secrets =
+      Enum.reject(org_secrets, fn secret ->
+        Enum.any?(agent_secrets, &(&1.name == secret.name))
+      end) ++ agent_secrets
```

Comment:
```suggestion
    secrets = Enum.uniq_by(agent_secrets ++ org_secrets, & &1.name)
```

Reply:
I was talking to @josevalim and we chose to perform `Enum.uniq_by/2` when the Livebook requests the Hub secrets (`handle_call`).